core-services/08-sysctl.sh

```diff
@@ -2,15 +2,19 @@
 
 if [ -x /sbin/sysctl -o -x /bin/sysctl ]; then
     msg "Loading sysctl(8) settings..."
+    mkdir -p /run/vsysctl.d
     for i in /run/sysctl.d/*.conf \
         /etc/sysctl.d/*.conf \
         /usr/local/lib/sysctl.d/*.conf \
-        /usr/lib/sysctl.d/*.conf \
-        /etc/sysctl.conf; do
+        /usr/lib/sysctl.d/*.conf; do
 
-        if [ -e "$i" ]; then
-            printf '* Applying %s ...\n' "$i"
-            sysctl -p "$i"
+        if [ -e "$i" ] && [ ! -e "/run/vsysctl.d/${i##*/}" ]; then
+            ln -s "$i" "/run/vsysctl.d/${i##*/}"
```

Comment:
could you just load the file here and touch the basename in `/run/vsycsctl.d` something that makes  is it really necessary to do this in a separate loop?

Reply:
the extra loop is needed so that the lexicographic priorities are evaluated equally across different configuration directories. e.g. a case like this:

```
/etc/sysctl.d/20-foo.conf
/usr/lib/sysctl.d/10-bar.conf
```

would result in the following boot-time generated dir:

```
/run/vsysctl.d/10-bar.conf
/run/vsysctl.d/20-foo.conf
```

Without the second loop, the etc file gets applied before the usr file.

Reply:
Ah ok, makes sense if this is how systems do it too.